domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/readmodel/PreparednessInSemester.kt

```diff
@@ -0,0 +1,31 @@
+package com.krzykrucz.elesson.currentlesson.preparedness.readmodel
+
+import arrow.core.Option
+import arrow.effects.IO
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.StudentMarkedUnprepared
+import com.krzykrucz.elesson.currentlesson.shared.AsyncOutput
+import com.krzykrucz.elesson.currentlesson.shared.ClassName
+import com.krzykrucz.elesson.currentlesson.shared.FirstName
+import com.krzykrucz.elesson.currentlesson.shared.SecondName
+import com.krzykrucz.elesson.currentlesson.shared.WholeNumber
+
+
+typealias WriteUnpreparednessInTheRegister = (StudentMarkedUnprepared) -> IO<StudentSubjectUnpreparednessInASemester>
+
+typealias GetStudentSubjectUnpreparednessInASemester = (StudentInSemester) -> AsyncOutput<StudentSubjectUnpreparednessInASemester, StudentInSemesterReadError>
+
+data class StudentInSemester(val className: ClassName, val firstName: FirstName, val secondName: SecondName) // todo add subject
```

Comment:
`//TODO` left, when will it be implemented?

Reply:
I will create another PR with that